src/openklant/components/klantinteracties/api/validators.py

```diff
@@ -166,3 +170,26 @@ def Rekeningnummer_exists(value):
         Rekeningnummer.objects.get(uuid=str(value))
     except Rekeningnummer.DoesNotExist:
         raise serializers.ValidationError(_("Rekeningnummer object bestaat niet."))
+
+
+class OptionalEmailValidator(EmailValidator):
+    """
+    EmailValidator for SoortDigitaalAdres that only attempts to validate if
+    `SoortDigitaalAdres` is `email
+    """
+
+    requires_context = True
+
+    def __call__(self, attrs: dict, serializer):
+        if (
+            get_field_value(serializer, attrs, "soort_digitaal_adres")
+            == SoortDigitaalAdres.email
+        ):
+            try:
+                super().__call__(get_field_value(serializer, attrs, "adres"))
+            except ValidationError as e:
+                # re-raise as field error
+                if serializer:
+                    raise exceptions.ValidationError({"adres": e.message})
+                else:
+                    raise ValidationError({"adres": e})
```

Comment:
Unfortunately I found no way around this, ideally I'd always raise a `django.core.exceptions.ValidationError`, but I wasn't able to raise that as a field specific error in rest_framework (it was always in the data as `nonFieldErrors`)

Reply:
Do you mean it shows up as `nonFieldErrors` in the admin? It might be related to calling the validator in the `clean` [method](https://github.com/maykinmedia/open-klant/pull/271/files#diff-4086aff50bf30b40ef88ec2f2299d06c4c266ae0e908cbe02c765e54f1079cacR13) (as opposed to the `clean_<field>`)

Reply:
No in the admin it works as expected, but the API returns it as a nonfielderror, regardless of whether I re-raise the error like this: `raise ValidationError({"adres": e})`

Reply:
hmmm, what about not using a validator (at the serializer level) but using the `validate` method on the specific serializer (as this validation is done for a very specific use case)?

Reply:
You're right, I've changed it so now I do not have to re-raise the error anymore